satpy/readers/agri_l1.py

```diff
@@ -0,0 +1,206 @@
+_COFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_CFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+_LOFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_LFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+
+
+class HDF_AGRI_L1(HDF5FileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDF_AGRI_L1, self).__init__(filename, filename_info, filetype_info)
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Load a dataset."""
+        logger.debug('Reading in get_dataset %s.', dataset_id.name)
+        file_key = ds_info.get('file_key', dataset_id.name)
+        lut_key = ds_info.get('lut_key', dataset_id.name)
+        data = self.get(file_key)
+        lut = self.get(lut_key)
+        if data.ndim >= 2:
+            data = data.rename({data.dims[-2]: 'y', data.dims[-1]: 'x'})
+
+        # convert bytes to string
+        data.attrs['long_name'] = data.attrs['long_name'].decode('gbk')
+        data.attrs['band_names'] = data.attrs['band_names'].decode('gbk')
+        if ds_info['file_type'] != 'agri_l1_4000m_geo':
+            data.attrs['center_wavelength'] = data.attrs['center_wavelength'].decode('gbk')
+
+        # calibration
+        calibration = ds_info['calibration']
+
+        if calibration == 'counts':
+            data.attrs['units'] = ds_info['units']
+            ds_info['valid_range'] = data.attrs['valid_range']
+            return data
+
+        elif calibration == 'reflectance':
+            logger.debug("Calibrating to reflectances")
+            # using the corresponding SCALE and OFFSET
+            cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
+            num_channel = self.get(cal_coef).shape[0]
+            if num_channel == 1:
+                # only channel_2, resolution = 500 m
+                slope = self.get(cal_coef)[:, 0].values
+                offset = self.get(cal_coef)[:, 1].values
+            else:
+                slope = self.get(cal_coef)[:, 0][int(file_key[-2:])-1].values
+                offset = self.get(cal_coef)[:, 1][int(file_key[-2:])-1].values
+            data = self.dn2reflectance(data, slope, offset)
+            ds_info['valid_range'] = (data.attrs['valid_range'] * slope + offset) * 100
+
+        elif calibration == 'brightness_temperature':
+            logger.debug("Calibrating to brightness_temperature")
+            # the value of dn is the index of brightness_temperature
+            data = self.calibrate(data, lut)
+            ds_info['valid_range'] = lut.attrs['valid_range']
+
+        data.attrs.update({'platform_name': self['/attr/Satellite Name'],
+                           'sensor': self['/attr/Sensor Identification Code'],
+                           'satellite_nominal_latitude': self['/attr/NOMCenterLat'],
+                           'satellite_nominal_longitude': self['/attr/NOMCenterLon'],
+                           'satellite_nominal_altitude': self['/attr/NOMSatHeight']})

```

Comment:
Unless I'm reading this wrong, I think the metadata added by @sfinkens meant to put these `satellite_nominal_*` parameters in a sub-dictionary called `orbital_parameters`.

Reply:
Yes, exactly! See also https://satpy.readthedocs.io/en/latest/readers.html#metadata .